NetworkWidget: Reorganise SSL options group box

sepalani opened the pull request:
This PR splits SSL options group box into 2 group boxes. One dedicated to dumping options, the other to the network security options. On top of that a combo box was added to select the network dump format and a button to open the dump folder.

![image](https://user-images.githubusercontent.com/7890055/107945407-02ad4c80-6fa9-11eb-93d4-12ed1dd68b84.png)
![image](https://user-images.githubusercontent.com/7890055/107945507-2bcddd00-6fa9-11eb-90da-15e2296a5998.png)

Ready to be reviewed & merged.

lioncash commented:
Seems quite largely sized relative to everything else in the SSL options to the point that it makes them look a little off kilter/unbalanced visually.

sepalani commented:
I added a size policy to prevent the button to fill the whole space:
![image](https://user-images.githubusercontent.com/7890055/80580291-5cde2080-8a1c-11ea-8922-03698ed67b8e.png)

leoetlino commented:
This still looks a bit odd IMO... also buttons are likely to be taller on other platforms, which would possibly result in inconsistent line height?

leoetlino commented:
So I tested this on Linux and the layout looks the same as on Windows, so no issues with line height. I'm still not a big fan of the UI but I can't think of any better alternative when you only have *one* button.

@lioncash any suggestions, or do you think the current layout is fine (now that the button width is fixed)?

sepalani commented:
@leoetlino @lioncash 
I changed this PR a bit since the PCAP PRs were merged. I added a combo box to select the network dump format (and greyed options when not available depending on the dump format).

![image](https://user-images.githubusercontent.com/7890055/107945507-2bcddd00-6fa9-11eb-90da-15e2296a5998.png)

leoetlino commented:
Why do we need separate options for SSL read/write, SSL read and SSL write? Wouldn't a simple SSL option suffice, considering the user can already choose to dump reads or writes or both?

sepalani commented:
The main issue I have, is that the SSL dump format is a legacy format that can only dump decrypted IOS SSL and was used to dump decrypted HTTPS messages in 2 text files. It was coded as 2 define which got replaced with 2 config options instead. So I have no easy way from a UI/UX point of view to make the differences between the None option (dump disabled) and an hypothetical "Binary SSL" option with both decrypted read and write disabled. It also can't dump raw SSL like PCAP.

In sum, if I have a to create a single Binary SSL option I would have to:
 - Make the distinction between the `None` option and `Binary SSL` with both decrypted read/write disabled
 - Or alternatively, Make sure both read/write can't be disabled or switch to None option when it's the case

leoetlino commented:
Okay, that makes sense to me.

leoetlino commented on Source/Core/DolphinQt/Debugger/NetworkWidget.cpp:
```diff
@@ -309,28 +337,98 @@ QGroupBox* NetworkWidget::CreateSSLContextGroup()
-
-  m_dump_ssl_read_checkbox = new QCheckBox(tr("Dump SSL read"));
-  m_dump_ssl_write_checkbox = new QCheckBox(tr("Dump SSL write"));
+  QGroupBox* dump_options_group = new QGroupBox(tr("Dump options"));
```
```suggestion
  auto* dump_options_group = new QGroupBox(tr("Dump options"));
```

to avoid repeating yourself when the type name is already written out on the right hand side

leoetlino commented on Source/Core/DolphinQt/Debugger/NetworkWidget.cpp:
```diff
@@ -309,28 +337,98 @@ QGroupBox* NetworkWidget::CreateSSLContextGroup()
-  m_dump_ssl_read_checkbox = new QCheckBox(tr("Dump SSL read"));
-  m_dump_ssl_write_checkbox = new QCheckBox(tr("Dump SSL write"));
+  QGroupBox* dump_options_group = new QGroupBox(tr("Dump options"));
+  QVBoxLayout* dump_options_layout = new QVBoxLayout;
```
```suggestion
  auto* dump_options_layout = new QVBoxLayout;
```

leoetlino commented on Source/Core/DolphinQt/Debugger/NetworkWidget.cpp:
```diff
@@ -309,28 +337,98 @@ QGroupBox* NetworkWidget::CreateSSLContextGroup()
+
+QGroupBox* NetworkWidget::CreateSecurityOptionsGroup()
+{
+  QGroupBox* security_options_group = new QGroupBox(tr("Security options"));
```
```suggestion
  auto* security_options_group = new QGroupBox(tr("Security options"));
```

leoetlino commented on Source/Core/DolphinQt/Debugger/NetworkWidget.cpp:
```diff
@@ -309,28 +337,98 @@ QGroupBox* NetworkWidget::CreateSSLContextGroup()
+QGroupBox* NetworkWidget::CreateSecurityOptionsGroup()
+{
+  QGroupBox* security_options_group = new QGroupBox(tr("Security options"));
+  QVBoxLayout* security_options_layout = new QVBoxLayout;
```
```suggestion
  auto* security_options_layout = new QVBoxLayout;
```

leoetlino commented on Source/Core/DolphinQt/Debugger/NetworkWidget.cpp:
```diff
@@ -309,28 +337,98 @@ QGroupBox* NetworkWidget::CreateSSLContextGroup()
+
+QComboBox* NetworkWidget::CreateDumpFormatCombo()
+{
+  QComboBox* combo = new QComboBox();
```
```suggestion
  auto* combo = new QComboBox();
```

sepalani commented:
@leoetlino Done.
